StyleCop.Analyzers/StyleCop.Analyzers/SpacingRules/SA1026CodeMustNotContainSpaceAfterNewKeywordInImplicitlyTypedArrayAllocation.cs

```diff
@@ -49,54 +49,31 @@ public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics
         /// <inheritdoc/>
         public override void Initialize(AnalysisContext context)
         {
-            context.RegisterSyntaxTreeActionHonorExclusions(this.HandleSyntaxTree);
+            context.RegisterSyntaxTreeActionHonorExclusions(HandleSyntaxTree);
         }
 
-        private void HandleSyntaxTree(SyntaxTreeAnalysisContext context)
+        private static void HandleSyntaxTree(SyntaxTreeAnalysisContext context)
         {
             SyntaxNode root = context.Tree.GetCompilationUnitRoot(context.CancellationToken);
             foreach (var token in root.DescendantTokens())
             {
-                switch (token.Kind())
+                if (!token.IsKind(SyntaxKind.NewKeyword))
                 {
-                case SyntaxKind.NewKeyword:
-                    this.HandleNewKeywordToken(context, token);
-                    break;
-
-                default:
-                    break;
+                    continue;
                 }
-            }
-        }
 
-        private void HandleNewKeywordToken(SyntaxTreeAnalysisContext context, SyntaxToken token)
-        {
-            if (token.IsMissing)
-            {
-                return;
-            }
-
-            if (!token.Parent.IsKind(SyntaxKind.ImplicitArrayCreationExpression))
-            {
-                return;
-            }
-
-            this.HandleDisallowedSpaceToken(context, token);
-        }
-
-        private void HandleDisallowedSpaceToken(SyntaxTreeAnalysisContext context, SyntaxToken token)
-        {
-            if (token.IsMissing || !token.HasTrailingTrivia)
-            {
-                return;
-            }
+                if (token.IsMissing || !token.Parent.IsKind(SyntaxKind.ImplicitArrayCreationExpression))
+                {
+                    continue;
+                }
 
-            if (!token.TrailingTrivia.First().IsKind(SyntaxKind.WhitespaceTrivia))
-            {
-                return;
+                SyntaxTriviaList trailingTriviaList = token.TrailingTrivia;
+                if (trailingTriviaList.Count > 0 &&
+                    (trailingTriviaList[0].IsKind(SyntaxKind.WhitespaceTrivia) || trailingTriviaList[0].IsKind(SyntaxKind.EndOfLineTrivia)))
```

Comment:
Based on the wording in [SA1022](http://www.stylecop.com/docs/SA1022.html) (and others), I'm under the impression that whitespace is just space and tab. However, for this rule I don't see any problem with implementing it to also not require any new lines between the tokens.